File: nodejs/access-token.js
// npm install jwt-simple
var jwt = require('jwt-simple');

module.exports = function() {
    var wdApiId = 'demo',
        wdApiSecret = 'demo';

    var payload = {
        iss: 'https://wikidocs.com/v1/apps/' + wdApiId,
        iat: Math.round((new Date().getTime()/1000)),
        exp: Math.round((new Date().getTime()/1000)) + 3600*24,
        sub: 'userId-123',
        access: {}
    };
    payload.access['/content-123'] = 'full';

    return jwt.encode(payload, wdApiSecret);
}

File: ruby/access-token.rb
# sudo gem install jwt
require 'JWT'

wdApiId = 'demo'
wdApiSecret = 'demo'

payload = {
	"iss" => 'https://wikidocs.com/v1/apps/' + wdApiId,
	"iat" => Time.now.getutc.to_i,
	"exp" => Time.now.getutc.to_i + 3600*24,
	"sub" => "userId-123",
	"access" => {
		"/content-123" => "full"
	}
};

accessToken = JWT.encode(payload, wdApiSecret)

print accessToken, "\n"